src/scaffoldmaker/meshtypes/meshtype_3d_heart1.py

```diff
@@ -447,17 +443,24 @@ def refineMesh(cls, meshrefinement, options):
             i += 1
 
     @classmethod
-    def generateMesh(cls, region, options):
+    def defineFaceAnnotations(cls, region, options, annotationGroups):
         """
-        Generate base or refined mesh.
-        :param region: Zinc region to create mesh in. Must be empty.
+        Add face annotation groups from the highest dimension mesh.
+        Must have defined faces and added subelements for highest dimension groups.
+        :param region: Zinc region containing model.
         :param options: Dict containing options. See getDefaultOptions().
-        :return: list of AnnotationGroup for mesh.
+        :param annotationGroups: List of annotation groups for top-level elements.
+        New face annotation groups are appended to this list.
         """
-        if not options['Refine']:
-            return cls.generateBaseMesh(region, options)
```

Comment:
I moved generateMesh to Scaffold_base; github has mistakenly compared generateMesh to a new function generateFaceAnnotations. I've separated out code for generating faces and annotations on them so that I can run it on either the original or the refined mesh.

Reply:
Ok noted